pkg/build/builder/dockerutil.go

```diff
@@ -156,40 +157,43 @@ func dockerRun(client DockerClient, createOpts docker.CreateContainerOptions, lo
 
 	containerName := containerNameOrID(c)
 
-	// Container was created, so we defer its removal.
-	defer func() {
+	// Container was created, so we defer its removal, and also remove it if we get a sigterm
+
+	removeContainer := func() {
 		glog.V(4).Infof("Removing container %q ...", containerName)
 		if err := client.RemoveContainer(docker.RemoveContainerOptions{ID: c.ID}); err != nil {
 			glog.Warningf("Failed to remove container %q: %v", containerName, err)
 		} else {
 			glog.V(4).Infof("Removed container %q", containerName)
 		}
-	}()
-
-	// Start the container.
-	glog.V(4).Infof("Starting container %q ...", containerName)
-	if err := client.StartContainer(c.ID, nil); err != nil {
-		return fmt.Errorf("start container %q: %v", containerName, err)
 	}
 
-	// Stream container logs.
-	logsOpts.Container = c.ID
-	glog.V(4).Infof("Streaming logs of container %q with options %+v ...", containerName, logsOpts)
-	if err := client.Logs(logsOpts); err != nil {
-		return fmt.Errorf("streaming logs of %q: %v", containerName, err)
-	}
+	return interrupt.New(nil, removeContainer).Run(func() error {
```

Comment:
Don't want to be called for bike-shedding, but `dockerRun` would be easier to read if we'd give a name to the function we pass to `interrupt...Run`.

```go
func removeContainer(name, id string) { ... }

func startWaitContainer(name, id string, logOpts ...) { ... } 

func dockerRun(client DockerClient, createOpts docker.CreateContainerOptions, ...) ... {
	// ...

	// Start and wait for the container to run, and ensure we call removeContainer even if
	// the program catches a SIGINT/SIGTERM/SIGQUIT signal.
	return interrupt.New(nil, removeContainer).Run(func() error {
		return startWaitContainer(...)
	})
	// or maybe:
	//startWaitContainer := func() error { return startWaitContainer(...) }
	//return interrupt.New(nil, removeContainer).Run(startWaitContainer)
}
```

I have in mind that using `interrupt` as a "better defer" is not obvious for readers familiar with conventional Go code, so it's good to make it clear what's going on for our future selves.